pkg/build/metrics/prometheus/metrics.go

```diff
@@ -46,23 +48,38 @@ var (
 )
 
 type buildCollector struct {
-	lister buildlister.BuildLister
+	lister     buildlister.BuildLister
+	isCreated  bool
+	createOnce sync.Once
+	createLock sync.RWMutex
 }
 
-// InitializeMetricsCollector calls into prometheus to register the buildCollector struct as a Collector in prometheus
-// for the terminal and active build metrics; note, in comparing with how kube-state-metrics integrates with prometheus,
-// kube-state-metrics leverages the prometheus.Registerer function, but it does not exist in the version of prometheus
-// vendored into origin as of this writing
+// IntializeMetricsCollector calls into prometheus to register the buildCollector struct as a
+// Collector in prometheus for the terminal and active build metrics.
 func IntializeMetricsCollector(buildLister buildlister.BuildLister) {
-	bc.lister = buildLister
-	// unit tests unearthed multiple (sequential, not in parallel) registrations with prometheus via multiple calls to new build controller
-	if !registered {
-		prometheus.MustRegister(&bc)
-		registered = true
+	if !bc.IsCreated() {
+		bc.lister = buildLister
+		legacyregistry.MustRegister(&bc)
 	}
 	klog.V(4).Info("build metrics registered with prometheus")
 }
 
+// Create satisfies the k8s metrics.Registerable interface. It is called when the metric is
+// registered with Prometheus via k8s metrics.
+func (bc *buildCollector) Create(v *semver.Version) bool {
+	bc.createOnce.Do(func() {
+		bc.createLock.Lock()
+		defer bc.createLock.Unlock()
+		bc.isCreated = true
+	})
+	return bc.IsCreated()
+}
```

Comment:
@openshift/openshift-team-monitoring ptal - this was pseudo-lifted from upstream's logic for initializing Prometheus core metric types.